wellcomeml/ml/bilstm.py

```diff
@@ -139,6 +139,8 @@ def residual_attention(x1):
         return model
 
     def fit(self, X, Y, embedding_matrix=None, *_):
+        X = np.array(X)
```

Comment:
This will fail when X is an sparse array (and when it is sparse, we don't want to convert it anyway!), so you probably need to check first whether it is a csr_matrix. 

Take a look at `tests/test_cnn.py:test_sparse`

Reply:
I am thinking if check they data type and it turns out to be a list then only we convert it ... ?

Reply:
Yes, that is one possible solution. I will let @nsorros give the final word!